src/main/java/org/sagebionetworks/bridge/services/Exporter3Service.java

```diff
@@ -934,4 +963,71 @@ private void exportUpload(String appId, String recordId) {
         LOG.info("Sent export request for app " + appId + " record " + recordId + "; received message ID=" +
                 sqsResult.getMessageId());
     }
+
+    // Export timeline  from Bridge to Synapse (Some researchers only have access to Synapse, not Bridge,
+    // so they need access to the Timeline information.).
+    public Exporter3Configuration exportTimelineForStudy(String appId, String studyId) throws BridgeSynapseException,
+            SynapseException, IOException{
+
+        // Get Timeline to export for the study.
+        Timeline timeline = schedule2Service.getTimelineForSchedule(appId,
+                studyService.getStudy(appId, studyId, true).getScheduleGuid());
+
+        // Check Synapse
+        synapseHelper.checkSynapseWritableOrThrow();
+
+        // Initiate Exporter3 for Study.
+        Exporter3Configuration exporter3Config = initExporter3ForStudy(appId, studyId);
+
+        // Export the study's Timeline to Synapse as a wiki page in JSON format.
+        JsonNode node = BridgeObjectMapper.get().valueToTree(timeline);
+
+//        File file = File.createTempFile("timelinefor" + studyId, ".txt");
+//        FileWriter fileWriter = new FileWriter(file);
+//        PrintWriter printWriter = new PrintWriter(fileWriter);
+//        printWriter.print(node.toString());
+//        printWriter.close();
+        PrintWriter pw = null;
+        File file = File.createTempFile("timelinefor" + studyId, ".txt");
+        try {
+            FileOutputStream fos = new FileOutputStream(file);
+            pw = new PrintWriter(fos);
+            pw.print(node.toString());
+            pw.close();
+            pw = null;
+        } finally {
+            if (pw != null) {
+                pw.close();
+            }
+        }
```

Comment:
You can make this even simpler by using Guava. First, open the File as a CharSink using https://javadoc.io/doc/com.google.guava/guava/latest/com/google/common/io/Files.html#asCharSink(java.io.File,java.nio.charset.Charset,com.google.common.io.FileWriteMode...)

Then, use write() https://javadoc.io/static/com.google.guava/guava/32.0.0-jre/com/google/common/io/CharSink.html#write(java.lang.CharSequence)